File: array/array1.go
package array

// RemoveDuplicates array sort
func RemoveDuplicates(nums []int) int {
	if len(nums) == 0 {
		return 0
	}
	i := 0
	for index, value := range nums {
		// if nums[index] not duplicate, rewrite number in nums[i]
		if value != nums[i] {
			i++
			nums[i] = nums[index]
		}
	}
	return i + 1
}

File: old/Array/array1.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function (nums) {
    let unique = 0;
  
    for (let i = 1; i < nums.length; i++) {
      if (nums[unique] !== nums[i]) {
        nums[++unique] = nums[i];
      }
    }
  
    // array 의 unique 값들만으로 구성된 부분까지의 길이를 return 해야함
    return unique + 1;
  };
